sql/hive/src/test/scala/org/apache/spark/sql/parquet/parquetSuites.scala

```diff
@@ -263,6 +356,22 @@ abstract class ParquetTest extends QueryTest with BeforeAndAfterAll {
     }
   }
 
+  Seq("partitioned_parquet_with_key_and_complextypes", "partitioned_parquet_with_complextypes").foreach { table =>
+    test(s"SPARK-5775 read struct from $table") {
+      checkAnswer(
+        sql(s"SELECT p,  structField.intStructField , structField.stringStructField FROM $table WHERE p = 1"),
+        (1 to 10).map { i => ((1, i, f"${i}_string"))}
+      )
+    }
+
+    test (s"SPARK-5775 read array from $table") {
+              checkAnswer(
+                sql(s"SELECT arrayField, p FROM $table WHERE p = 1"),
+                (1 to 10).map { i => ((1 to i,1))}
```

Comment:
Thanks for the review !

I will be checking further commits against https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide . Would kindly indicate if tool to automatically check for the type of issues you raised ?

A.